conans/client/conf/detect.py

```diff
@@ -169,18 +177,11 @@ def _detect_compiler_version(result, output, profile_path):
 
 
 def _detect_os_arch(result, output):
-    architectures = {'i386': 'x86',
```

Comment:
Same comment as above, if it is not related with the AIX we need you to open a different PR. I can see here some changing behavior like the  `sun4v` is not well managed anymore.

Reply:
this is a duplication of the code in detected_architecture which leads to different behavior between those methods or duplicate code. sun4v is the only thing missing in the detected_architecture handling which i will try to add there

Reply:
Yes, we know and understand but please open a new PR with this refactor, we agree that it is pretty duplicated but the refactor is changing the behavior and it is not needed for supporting the AIX setting. As I said, for example `sun4v` key is not considered in the `detected_architecture`. We always try to keep the PRs as lean as possible, we have a compatibility commitment and we have to try hard not to change any possible breaking behavior.